syntax = "proto3";

package kvstore;

service LBNodeComm {
    rpc SendHeartBeat(stream HeartBeatRequest) returns (stream HeartBeatReply) {}
    rpc AssertLeadership(AssertLeadershipRequest) returns (AssertLeadershipReply) {}
}

message HeartBeatRequest {
    uint32 identity = 1;
    string ip = 2;
}

message HeartBeatReply {
    repeated NodeData node_data = 1;
}

message NodeData {
    string ip = 1;
    uint32 identity = 2;
}

message AssertLeadershipRequest {
    string leader_ip = 1;
}

message AssertLeadershipReply {
    repeated FollowerIP follower_ip = 1;
}

message FollowerIP {
    string ip = 1;
}